syntax = "proto3";

package proto_catalogue;

import "transport_router.proto";
import "map_renderer.proto";

message Stop{
  string name = 1;
  double latitude = 2;
  double longitude = 3;
}

message Distance {
  string stop_name_from = 1;
  string stop_name_to = 2;
  uint32 distance = 3;
}

message Route{
  string name = 1;
  repeated string stop_name = 2;
  bool is_roundtrip = 3;
}

message TransportCatalogue {
  proto_renderer.RenderSetting render_settings = 1;
  proto_router.Router router = 2;
  repeated Stop all_stops = 3;
  repeated Distance all_distances = 4;
  repeated Route all_routes = 5;
}



